### PR TITLE
Add ClusterResourceOverridesEnabled for console config

### PR DESCRIPTION
**Pull request by spadgett**
We plan to have one flag in the console config that says whether ClusterResourceOverrides are enabled. The console will hide the CPU request, CPU limit, and memory request fields if the flag is true.

Regardless of whether the flag is set, we'll show if a value was changed using the annotation from https://github.com/openshift/origin/pull/18007. If a deployment has pods with changed value, we can also show that on the deployment page.

/assign @deads2k 
/cc @jwforres @smarterclayton

### CODE REVIEW

File: webconsole/v1/types.go [hunk 1]
```diff
@@ -42,6 +42,10 @@ type ClusterInfo struct {
 	// LogoutPublicURL is an optional, absolute URL to redirect web browsers to after logging out of the web
 	// console. If not specified, the built-in logout page is shown.
 	LogoutPublicURL string `json:"logoutPublicURL" protobuf:"bytes,5,opt,name=logoutPublicURL"`
+
+	// ClusterResourceOverridesEnabled indicates that the cluster is configured for overcommit and the web console
+	// should hide request and limit fields that will be overridden in its editors
```
Comment (jwforres) [inline webconsole/v1/types.go]: Should we be more clear in this comment that we assume this means all cluster resource overrides are in use and all memory and cpu fields will be hidden except for the memory limit?

Comment (spadgett) [reply webconsole/v1/types.go]: Yeah I'll update it with more detail

**Comment by spadgett** [comment]
Updated

**Comment by spadgett** [comment]
PR with console changes: https://github.com/openshift/origin-web-console/pull/2671

**Comment by spadgett** [comment]
@deads2k any objections?

**Comment by spadgett** [comment]
Updated

**Comment by jwforres** [comment]
LGTM

**Comment by deads2k** [comment]
Thanks. Will merge on green.

**Pull request merged**